### ISSUE TITLE
sbt 1.3.3 project depends on some non-local files

### ISSUE THREAD

**Issue by agluszak** [sbt/sbt#5208]
## steps
sbt version: 1.3.3

Minimal, Reproducible Example:
```
git clone https://github.com/agluszak/sbt-bug.git
cd sbt-bug
docker run -it -v $PWD:/workspace iterators/docker-git-sbt:1.3.3_2.12.9 bash
# inside docker
cd workspace
sbt compile
exit
docker run -it -v $PWD:/workspace iterators/docker-git-sbt:1.3.3_2.12.9 bash
# inside docker
cd workspace
sbt test
```
## problem
```
bash-5.0# sbt test
[info] Loading settings for project workspace-build from plugins.sbt ...
[info] Loading project definition from /workspace/project
/workspace/build.sbt:7: error: not found: value wartremoverWarnings
wartremoverWarnings ++= Warts.all
^
/workspace/build.sbt:7: error: not found: value Warts
wartremoverWarnings ++= Warts.all
                        ^
[error] Type error in expression
```
Sbt project with plugin (WartRemover) apparently depends on some non-local files (outside the project folder). We discovered it after sbt version update (from 1.1.5) on our CI. We used to copy /target folder between compile and test stages, but after update it broke. It turns out that something else outside the project folder is needed, because moving the whole folder between stages (as demonstrated above) doesn't help.

## expectation
It should compile just fine.

## notes
In the beginning we updated to 1.3.2, but I checked issues closed in 1.3.3 and #5109 seemed to be connected, but updating to 1.3.3 didn't change anything.

**Comment by eed3si9n** [comment]
@agluszak Thanks for the report with repro project!

**Comment by eed3si9n** [comment]
I haven't tested it, but there are a bunch of things that might need `~/.sbt`.
- global (machine-wide) settings
- global (machine-wide) plugins
- compiler bridge cache
- sbt artifact cache

**Comment by agluszak** [comment]
We use deafult global settings, no global plugins (WartRemover is explicitly added in project/plugins.sbt). And what's most important - it used to work in sbt 1.1

**Comment by lustefaniak** [comment]
We have the same problem, we have multiple drone steps, one running tests, another depending on target staging sbt-native-packager docker image - that one fails with not found keys. This used to work in 1.2.8.

I tried all 1.3.x versions and they all have same problem. Might be Drone CI specific issue, but if it used to work in all older sbt versions (we started using drone with sbt 0.13) and now stopped it has to be something sbt 1.3.x specific.

@agluszak are you using custom sbt/coursier cache locations?
